diff --git a/onnxruntime/core/mlas/lib/intrinsics/avx512/quantize_avx512f.cpp b/onnxruntime/core/mlas/lib/intrinsics/avx512/quantize_avx512f.cpp
index 47d2dce45c..162353a8ef 100644
--- a/onnxruntime/core/mlas/lib/intrinsics/avx512/quantize_avx512f.cpp
+++ b/onnxruntime/core/mlas/lib/intrinsics/avx512/quantize_avx512f.cpp
@@ -1,3 +1,7 @@
+#if !defined(__clang__) && defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#endif
 /*++
 
 Copyright (c) Microsoft Corporation. All rights reserved.
